doc/ERROR_CONCEPT.md

```diff
@@ -0,0 +1,250 @@
+# Error Message Concept
+
+The purpose of this file is solely to better comment on in a Pull Request and gather feedback.
+
+## Findings/ Motivation
+
+1. Too many errors in the [specification](src\error\specification) (210 errors/warnings)
+2. Many duplicate or redundant errors
+3. Description/ Reason are very similar
+4. Potential file splitting (1327 LOC)
+5. Verbose error message
+
+## Some facts
+
+1. Some errors have the same text but are duplicated because one is `warning` and the other one is `error`/`fatal` 
+(#153 + #154, #128 + #129, #120 + #41, #44 + 45 , #66 + #67, etc.)
+2. Some errors might be duplicated just because they belong to different plugins (or the dev neglected to lookup a similar in the specification) 
+(#29 + #57 + #72 + #169, #10 + #61, etc.)
+3. Many errors are very similar or could be summarized and are differentiated via the reason field 
+(#54 + #55, #31 - #33, #93 + #7 + #17, #74 + #75, #77 + #10 + #61, etc.)
+4. Some errors are marked as unused (#22 + #48 + #58 + #66 + #67 + #68)
+5. In many cases the `Description` is not very relevant at all (#199, #196, #187, etc.)
+
+## Evaluation
+
+From now on I will take the following error message as example:
+
+```
+1. The command kdb set failed while accessing the key database with the info:
+2. Sorry, the error (#121) occurred ;(
+3. Description: validation failed
+4. Reason: Validation of key "<key>" with string "<value>" failed.
+5. Ingroup: plugin
+6. Module: enum
+7. At: ....../src/plugins/enum/enum.c:218
+8. Mountpoint: <parentKey>
+9. Configfile: ...../<file>.25676:1549919217.284067.tmp
+```
+
+The lines are numbered and will be referred in the following sections.
+
+Generally I would suggest that this error message is way too much information for a user. We can agree on the fact 
+that #4 is by far the most relevant line.
+
+### Line 9: Configfile
+
+Reasons to keep it:
+* If users want to see where the concrete file is located 
+
+Reasons against it:
+* Very verbose line
+* I highly doubt that most common users will require this information (unnecessary)
+
+Suggestions:
+Remove it completely. If users want to know the location they still can use `kdb file` on the concrete setting.
```

Comment:
The information was added on request, so I would not remove it completely. Maybe we disable it by default and only show it on request (-v for cmd-line).